src/app/functions/server/featureFlags.test.ts

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// import { it, expect, jest } from "@jest/globals";
+
+import { getFeatureFlagByName } from "../../../db/tables/featureFlags";
+jest.mock("../../../db/tables/featureFlags");
+
+jest.mock("next-auth", () => ({
+  getServerSession: jest.fn(() => {
+    return { user: { email: "test@example.com" } };
+  }),
+}));
+
+jest.mock("../../api/utils/auth", () => ({
+  authOptions: jest.fn(),
+}));
+
+jest.mock("../../../db/tables/featureFlags", () => ({
+  getFeatureFlagByName: jest.fn(),
+}));
+
+import { isFlagEnabled } from "./featureFlags";
+
+beforeEach(() => {
+  jest.resetModules();
+});
+
+it("throws if there is no valid session", async () => {
+  await expect(isFlagEnabled("testFlag")).resolves.toBe(false);
+});
+
+it("returns false if flag not provided or nonexistent", async () => {
+  //@ts-ignore TS should prevent, but let's test at runtime anyway.
```

Comment:
You can even turn this into a test to verify that the type definition also disallows it at build time:

```suggestion
  // @ts-expect-error TS should prevent, but let's test at runtime anyway.
```

See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments